File: 프로그래머스/lv0/120811. 중앙값 구하기/중앙값 구하기.js
function solution(array) {
    let result = array.sort( (a, b) => {return a-b;});
    let answer = array[Math.floor(array.length/2)]
    return answer;
}
/*
    접근법 :
    sort() 를 활용해 오름차순 정렬 시키고, 중앙에 있는 인덱스 값을 뽑아 반환해주면 되지 않을까?
    매개변수의 오름차순 정렬 -> result
    중앙에 있는 인덱스 값 구하기 -> answer

    **주의 :
    매개변수로 들어온 배열은 홀수 라서 중앙에 있는 인덱스를 뽑아올 때,
    Math.floor() 함수를 이용해 Ex) 3.5 -> 3 정수로 잘라줘야 한다.
*/

File: 프로그래머스/lv0/120811. 중앙값 구하기/중앙값 구하기.java
import java.util.Arrays;
class Solution {
    public int solution(int[] array) {
        Arrays.sort(array);
        int answer = array[(int)Math.floor(array.length/2)];
        return answer;
    }
}
/*
    접근법 :
    JS 와는 다르게 자바는 역시 타입 때문에 생각이 좀 길어진다.
    내장함수가 없으므로, Arrays 를 임포트해 .sort() 메소드 사용
    Math.floor() 를 이용해 중앙의 인덱스 찾아내기 double 타입으로 반환되므로, int로 형변환

    **주의 :
    Math 클래스의 모든 메소드는 클래스 메소드(static method)이므로,
    객체를 생성하지 않고도 바로 사용할 수 있다.
    이러한 Math 클래스는 java.lang 패키지에 포함되어 제공
*/